proxy/proxy.go

```diff
@@ -848,6 +848,10 @@ func (p *Proxy) makeBackendRequest(ctx *context, requestContext stdlibcontext.Co
 	if endpoint != nil {
 		endpoint.Metrics.IncInflightRequest()
 		defer endpoint.Metrics.DecInflightRequest()
+
+		entry := p.registry.GetMetrics(endpoint.Host)
+		entry.IncInflightRequest()
+		defer entry.DecInflightRequest()
```

Comment:
We should also count inflight requests for non-loadbalanced backends. k8s dataclient uses network backend when service has a single endpoint https://github.com/zalando/skipper/blob/07dee9e2c9876937147cb6bc9e467544ba31ce21/dataclients/kubernetes/ingressv1.go#L137-L149

Reply:
From my side, it looks like it's better to return `hostname:port` from `mapRequest` and don't think what backend type it is at all.